some

suntsovpavel opened the pull request.

Commit:
first edition

Commit:
some edit...

suntsovpavel committed:
Update Human.java

some edit...

Commit:
some edit...

Commit:
some edit...

suntsovpavel committed:
Update Human.java

suntsovpavel committed:
Update Human.java

Liberate520 commented on src/FamilyTree.java:
```diff
@@ -0,0 +1,15 @@
+import java.util.List;
+
+public class FamilyTree {
+    private List<Human> family = new ArrayList<>();
```
инициализацию лучше делать в конструкторе класса

Liberate520 commented on src/Human.java:
```diff
@@ -0,0 +1,56 @@
+import java.util.ArrayList;
+
+public class Human {
+    private String Name;
```
имена полей с маленькой буквы

Liberate520 commented on src/Human.java:
```diff
@@ -0,0 +1,56 @@
+    private List<Human> childrens = new ArrayList<>();    
+    private LocalDate dateBirth, dateDeath;
+    private Gender gender;
+    Human(String Name, LocalDate dateBirth, Gender gender){
```
конструктор должен быть публичным

Liberate520 commented on src/Human.java:
```diff
@@ -0,0 +1,56 @@
+        return true;
+    }
+    public List<Human> getChildrens() { return childrens; }
+    public boolean setChild(Human child){
```
addChild

Commit:
eliminate comments homework 1

Commit:
add class Marriage

Commit:
add SaveLoad, FileHandler

Commit:
some edit

Liberate520 commented on src/FileHandler.java:
```diff
@@ -0,0 +1,22 @@
+import java.io.*;
+
+public class FileHandler implements SaveLoad, Serializable {
```
Объект класса FileHandler мы сохранять не собираемся, значит интерфейс Serializable для него не актуален

Liberate520 commented on src/FileHandler.java:
```diff
@@ -0,0 +1,22 @@
+public class FileHandler implements SaveLoad, Serializable {
+    //Этот класс выполняет сохранение в файл и загрузку из файла
+
+    public void SaveTo(Object obj, Object[] s) throws IOException {
```
имена методов с маленькой буквы

Liberate520 replied:
Какой то странный способ указать путь к файлу через массив

Liberate520 commented on src/Main.java:
```diff
@@ -0,0 +1,39 @@
+
+        //Выполняем запись в файл и последующее чтение
+        FileHandler filehandler = new FileHandler();
+        filehandler.SaveTo(tree, new String[] {"tree.txt"});
```
поменяйте второй аргумент на обычный стринг

Liberate520 commented on src/Marriage.java:
```diff
@@ -0,0 +1,59 @@
+    private LocalDate startDate, endDate;
+    private Human wife, husband;
+    private static final int ageAdulthood = 18;     //возраст совершеннолетия
+    private boolean isError = false;
```
наверное это результат работы какого то метода?

Liberate520 commented on src/Human.java:
```diff
@@ -0,0 +1,92 @@
+    public Marriage getMarriage() {
+        return marriage;
+    }
+    // TO DO как-то сделать этот метод доступным только классу Marriage
+    public void setMarriage(Marriage marriage) {
+        this.marriage = marriage;
+    }
```
Странная идея. Вы хотите чтобы только какой то определенный тип данных мог взаимодействовать с этой структурой. При этом в качестве аргумента передается именно этот тип данных

Liberate520 replied:
Думаю, что человек не должен нести в себе всю эту информацию, максимум ссылку на супругу. Данные о регистрации должны храниться где то отдельно и при желании можно было бы получать эти данные из репозитория по имени супруга например. Дворец бракосочетания это не часть человека, но может я и ошибаюсь. В любом случае вы это можете сделать только если в проекте появится пакетная структура и они будут лежать в одном пакете и тогда можно сделать метод с модификатором доступа дефолтным, то есть когда пусто

Commit:
Устранение замечаний 2

Commit:
add HumanGroup, HumanIterator

Commit:
ready homework

Commit:
add FileHandler

Commit:
some edit...

Commit:
some edit...

Commit:
ДЗ 4

Commit:
edit..

suntsovpavel committed:
Update FamilyTree.java

suntsovpavel committed:
Update FamilyTree.java

Commit:
edit Service and FamilyTree names methods

Commit:
add ItemFamilyTree
add generic for Marriage

suntsovpavel committed:
Update Marriage.java

suntsovpavel committed:
Update FamilyTree.java

Commit:
add try-block in main

Commit:
add try-block in main

Liberate520 commented on src/family_tree/group/GroupAgedNamed.java:
```diff
@@ -0,0 +1,13 @@
+
+public class GroupAgedNamed<E extends GroupItemAgedNamed> extends Group<E> implements Iterable<E> {
+    public void sortByName(){
+        listItem.sort(new ComparatorByName<>());
```
че т я не понимаю. А откуда здесь доступ к переменной listItem? Думаю тут все красным подчеркнуто)) она тут нигде не объявлена.

suntsovpavel replied:
class GroupAgedNamed наследует класс  Group

Liberate520 commented on src/family_tree/group/Group.java:
```diff
@@ -0,0 +1,31 @@
+import java.util.List;
+
+public class Group<E> implements Iterable<E>, Serializable {
+    protected List<E> listItem;
```
окей, не заметил, но protected все равно не нужен, достаточно не указывать никакого модификатора доступа, так как класс, которому это поле нужно находится в том же пакете

Commit:
ДЗ 5

Liberate520 commented on src/family_tree/presenter/Presenter.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void printInfoTree() {
+        System.out.println(service.getInfoAll());
```
работать с консолью должна view составляющая проекта. в презентере должно было быть view.print("какой то текст")

Liberate520 commented on src/family_tree/model/ServiceFamilyTree.java:
```diff
@@ -0,0 +1,180 @@
+    public FamilyTree getTree(){
+        return tree;
+    }
+
```
добавить методы сохранения и загрузки

Commit:
Устранение замечаний:
1. В Presenter используем view.print() вместо System.out.println
2. Классу FamilyTree возвращен джейнерик

Commit:
Устранение замечаний:
3. Добавлены функции записи и чтения дерева из файла

Commit:
Принцип 4 Разделение интерфейсов. Вместо единого интерфейса SaveLoad два раздельных SaveTo, LoadFrom

Commit:
Принцип 5. Реализуем взаимодействие ServiceFamilyTree с классами чтения и записи в файл FileLoader, FileSaver через интерфейсы SaveTo, LoadFrom

Commit:
Убираем из ServiceFamilyTree методы, не задействованные в MVP

Commit:
Принцип 5. Теперь класс FamilyTree использует гурппы Group, GroupItemFamilyTree не напрямую, а через интерфейсы Collectable, CollectableItemFamilyTree

Commit:
Принцип 1. Разделение функций. В классе Human заменяем реализацию метода getInfo использованием интерфейса Informer<Human>. Соответственно добавляем класс InformerHuman с конкретной реализацией вывода инфо

Commit:
Принцип 1. Разделение функций. В классе Marriage заменяем реализацию метода getInfo использованием интерфейса Informer<Marriage>. Соответственно добавляем класс InformerMarriage с конкретной реализацией вывода информации

Commit:
add ReportableFamilyTree.java
